### PR TITLE
ixed the issue when publishing stats for default version web apps

### PR DESCRIPTION
**Pull request by lakshani**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java [hunk 1]
```diff
@@ -112,9 +113,15 @@ public boolean handleRequest(MessageContext mc) {
...
-            String api_version = api + ":" + version;
+            WebApp webApp = null;
+            if (version != null) {
+                webApp = getWebApp(context, version);
+            } else {
+                webApp = getNonVersionedWebApp(context, tenantDomain);
+                version = webApp.getId().getVersion();
+            }
+            String appName = webApp.getId().getApiName();
+            String api_version = appName + ":" + version;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:124]: api_version >> appNameWithVersion

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java [hunk 2]
```diff
@@ -356,6 +362,54 @@ public WebApp getWebApp(String context,String version) throws AppManagementExcep
...
+     * Get WebApp for default version web apps.
+     * @param context
+     * @param tenantDomain
+     * @return
+     * @throws AppManagementException
+     */
+    public WebApp getNonVersionedWebApp(String context, String tenantDomain) throws AppManagementException {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:375]: replace with meaningful variable name

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java [hunk 3]
```diff
@@ -356,6 +362,54 @@ public WebApp getWebApp(String context,String version) throws AppManagementExcep
...
+            int tenantId = UsageComponent.getRealmService().getTenantManager().getTenantId(tenantDomain);
+            ps.setString(1,context);
+            ps.setInt(2,tenantId);
+            rs = ps.executeQuery();
+            while(rs.next()){
+                String webAppName = rs.getString("APP_NAME");
+                String provider = rs.getString("APP_PROVIDER");
+                String trackingCode = rs.getString("TRACKING_CODE");
+                String version = rs.getString("PUBLISHED_DEFAULT_APP_VERSION");
+                APIIdentifier apiIdentifier = new APIIdentifier(provider,webAppName,version);
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:395]: get column name from appMconstants.

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java (hunk 1 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:124]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/d4eba04341eb91c4b68b50cbb78987715b1a657d

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java (hunk 2 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:375]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/d4eba04341eb91c4b68b50cbb78987715b1a657d

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java (hunk 3 above)
Comment (lakshani) [reply components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:395]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/d4eba04341eb91c4b68b50cbb78987715b1a657d

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java [hunk 4]
```diff
@@ -356,6 +362,54 @@ public WebApp getWebApp(String context,String version) throws AppManagementExcep
...
+                "APM_APP_DEFAULT_VERSION.APP_PROVIDER=APM_APP.APP_PROVIDER WHERE APM_APP.CONTEXT=? AND APM_APP" +
+                ".TENANT_ID=?";
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            ps = conn.prepareStatement(sqlQuery);
+            int tenantId = UsageComponent.getRealmService().getTenantManager().getTenantId(tenantDomain);
+            ps.setString(1,context);
+            ps.setInt(2,tenantId);
+            rs = ps.executeQuery();
+            while(rs.next()){
```
Comment (Kishanthan) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:390]: Not formatted

Comment (lakshani) [reply components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:390]: Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/207d3b1c14192fee91f6d02b39c54f42bdf46e92

File: components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java [hunk 5]
```diff
@@ -356,6 +362,54 @@ public WebApp getWebApp(String context,String version) throws AppManagementExcep
...
     }
 
+    /**
+     * Get WebApp for default version web apps.
+     * @param context
+     * @param tenantDomain
+     * @return
+     * @throws AppManagementException
+     */
+    public WebApp getNonVersionedWebApp(String context, String tenantDomain) throws AppManagementException {
```
Comment (lakshani) [inline components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java:372]: Formatting and java doc comment issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files

**Pull request merged**